File: hw1/main.pl
/*
This code is from Rosettacode.org
http://rosettacode.org/wiki/FizzBuzz
*/

fizzbuzz :-
        foreach(between(1, 100, X), print_item(X)).
 /*
Precondition: Function argument X for print_item
Strategy = uses matching strategy just like if, elif, elif ... 
0 is X mod 15 == (if x mod 15 == 0)
 */
print_item(X) :- 
        (  0 is X mod 15
        -> print('FizzBuzz')
        ;  0 is X mod 3
        -> print('Fizz')
        ;  0 is X mod 5
        -> print('Buzz')
        ;  print(X)
        ),
        nl.

File: hw1/main.go
package main
 
import "fmt"
 
func main() {
    for i := 1; i <= 100; i++ {
        if i%3==0 {
            fmt.Printf("Fizz")
            if i%5==0{
                fmt.Printf("Buzz")
            }
        } else if i%5==0 {
            fmt.Printf("Buzz")
        } else {
            fmt.Printf("%v",i)
        }
        fmt.Printf("\n")
    }
}